Output binding for SQS

juanmjacobs opened the pull request:
I have yet to test this, but I figured I'd make a draft pull request for you to review if you'd like to.
I think this would resolve https://github.com/laveeshb/azure-function-extensions-net/issues/25

laveeshb commented:
Absolutely. I'll take a look.

laveeshb commented on Amazon.SQS/src/Extensions.SQS/Extensions.SQS.csproj:
```diff
@@ -11,7 +11,7 @@
     <AssemblyName>Azure.Functions.Extensions.SQS</AssemblyName>
     <RootNamespace>Azure.Functions.Extensions.SQS</RootNamespace>
-    <Version>1.0.1</Version>
+    <Version>1.0.2</Version>
```
I'll take care of updating the version when I publish the package.

laveeshb commented on Amazon.SQS/src/Extensions.SQS/Collector/SqsQueueAsyncCollector.cs:
```diff
@@ -0,0 +1,37 @@
+		}
+		public async Task AddAsync(SqsQueueMessage item, CancellationToken cancellationToken = new CancellationToken())
+		{
+			await AmazonSQSClient.SendMessageAsync(item.QueueUrl ?? this.TriggerParameters.QueueUrl, item.Body, cancellationToken);
```
All the parameters should be coming from one place IMO. I'd say just use the TriggerParameters for Url (after all client has to specify AWS credentials there). QueueMessage would include just the body and properties of the message.

laveeshb replied:
I see why you did it this way. This would allow clients to push messages to a different queue than the one they used in the trigger. I can see the benefit of that.

laveeshb commented on Amazon.SQS/src/Extensions.SQS/Collector/SqsQueueMessage.cs:
```diff
@@ -0,0 +1,8 @@
+﻿namespace Azure.Functions.Extensions.SQS.Collector
+{
+	public class SqsQueueMessage
```
I would also add initial visibility delay property. I can see clients having a use for it.

juanmjacobs replied:
I've added a binding for the native amazon sqs sdk message so as to have access to all of the supported options. 
But I also left this class for clients who only care about the body they are pushing.

laveeshb commented on Amazon.SQS/src/Extensions.SQS/Collector/SqsQueueAsyncCollector.cs:
```diff
@@ -0,0 +1,37 @@
+	public class SqsQueueAsyncCollector : IAsyncCollector<SqsQueueMessage>
+	{
+		private AmazonSQSClient AmazonSQSClient { get; }
+		private SqsQueueTriggerAttribute TriggerParameters { get; }
```
There is just one caveat here. This would limit use of the output bindings to only those functions that are using queue as a trigger as well. It is very well possible for a function to get triggered by some other mechanism but want to queue messages for another consumer.
That would be a different work item though.

laveeshb commented on Amazon.SQS/src/Extensions.SQS/Collector/SqsQueueAsyncCollector.cs:
```diff
@@ -0,0 +1,37 @@
+			this.AmazonSQSClient = AmazonSQSClientFactory.Build(triggerParameters);
+
+		}
+		public async Task AddAsync(SqsQueueMessage item, CancellationToken cancellationToken = new CancellationToken())
```
please add the new operation to readme with a sample

juanmjacobs commented:
Hi! Great, thank you for your comments. I'll try and get to your suggestions this week!

juanmjacobs commented:
Hey @laveeshb! I've finished working on this. I've tested it and it works. 
I made some refactors, added the README section as per your comment. And I also added a new attribute for the output messages, so it can be used in combination for any trigger.
